src/main/java/org/wso2/carbon/securevault/azure/AzureKeyVaultSecretCallbackHandler.java

```diff
@@ -0,0 +1,161 @@
+
+    private static final String ALIAS = "alias";
+    private static final String CONFIG_FILE_PATH = CarbonUtils.getCarbonConfigDirPath() + File.separator +
+            "security" + File.separator + "secret-conf.properties";
+    private static final String PASSWORD = "password";
+    private static final Log log = LogFactory.getLog(AzureKeyVaultSecretCallbackHandler.class);
+    private static String keyStorePassword;
+    private static String privateKeyPassword;
+
+    /**
+     * Handles single secret callback.
+     *
+     * @param singleSecretCallback a single secret callback.
+     */
+    @Override
+    protected void handleSingleSecretCallback(SingleSecretCallback singleSecretCallback) {
+
+        if (keyStorePassword == null && privateKeyPassword == null) {
+            boolean sameKeyAndKeyStorePass = true;
+            String keyPassword = System.getProperty(KEY + DOT + PASSWORD);
+
+            if (keyPassword != null && keyPassword.trim().equals("true")) {
+                sameKeyAndKeyStorePass = false;
+            }
+
+            readPasswordFromKeyVault(sameKeyAndKeyStorePass);
+            readPasswordThroughConsole(sameKeyAndKeyStorePass);
+        }
+
+        if (singleSecretCallback.getId().equals(IDENTITY + DOT + KEY + DOT + PASSWORD)) {
+            singleSecretCallback.setSecret(privateKeyPassword);
+        } else {
+            singleSecretCallback.setSecret(keyStorePassword);
+        }
+    }
+
+    /**
+     * Reads keystore and primary key passwords from Azure Key Vault.
+     *
+     * @param sameKeyAndKeyStorePass flag to indicate whether the keystore and primary key passwords are the same.
+     */
+    @SuppressFBWarnings("PATH_TRAVERSAL_IN")
+    private void readPasswordFromKeyVault(boolean sameKeyAndKeyStorePass) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Reading Carbon Secure Vault configuration properties from file.");
+        }
+
+        InputStream inputStream = null;
+        Properties properties = new Properties();
+
+        try {
+            inputStream = new FileInputStream(CONFIG_FILE_PATH);
+            properties.load(inputStream);
+        } catch (Exception e) {
+            throw new SecureVaultException("Error while loading configurations from " + CONFIG_FILE_PATH, e);
+        } finally {
+            try {
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                log.warn("Error closing input stream of configuration file.");
+            }
+        }
+
+        AzureKeyVaultRepository.authenticateToKeyVault(properties);
+        AzureKeyVaultRepository azureKeyVaultRepository = new AzureKeyVaultRepository();
+
+        String keyStoreAlias = properties.getProperty(KEY + STORE + DOT + IDENTITY + DOT + STORE + DOT + ALIAS);
+        String privateKeyAlias = properties.getProperty(KEY + STORE + DOT + IDENTITY + KEY + STORE + DOT + ALIAS);
```

Comment:
Recheck the property construction. Seems it is wrong.

Reply:
There was a typo. Fixed now.